lib/lambda_ethereum_consensus/validator/validator_set.ex

```diff
@@ -73,11 +75,12 @@ defmodule LambdaEthereumConsensus.ValidatorSet do
   def notify_head(set, slot, head_root) do
     Logger.debug("[ValidatorSet] New Head", root: head_root, slot: slot)
     epoch = Misc.compute_epoch_at_slot(slot)
+    head_state = fetch_target_state_and_go_to_slot(epoch, slot, head_root)
```

Comment:
After rethinking, this is unnecessary here. We need it for attestations, but we can fetch it at the attest funtion where we already know if there are validators to attest or not.

Reply:
Done